### ISSUE TITLE
[TW-24] incorrect use of xterm colorization escape sequences

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#86]
_steve rader on 2012-11-23T17:02:55Z says:_ 

 xterm-256color colors 0 through 15 are dependent on the background color.
the default task colorization uses them, and appears to assume the use of something like green on back.
thus the default colorization is really, um, bad when using (the fairly commmon?) back on white.
in my own code, i resolve this problem by e.g...
replacing color0 with color16 (relative black with absolute black)
replacing color15 with color231 (relative white with absolute white)
in other words, the default use of colors 0 through 15 need to be replaced equiv. colors >15

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2012-11-23T17:02:55Z
Modified: 2014-12-26T15:41:45Z
```

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2012-12-06T21:12:32Z says:_ 

 > xterm-256color colors 0 through 15 are dependent on the background color.

Is that not true of any color?

**Comment by taskwarrior** [comment]
_steve rader on 2012-12-07T15:37:25Z says:_ 

 
> > xterm-256color colors 0 through 15 are dependent on the background color.
> Is that not true of any color?

No, not at all.
I just explored this more thoroughly; my characterization was wrong...
Background/foreground colors don't matter.  It's normal video vs reverse video.
In "normal video", the 0-15 colors are always the same.
In "reverse video", the colors 1-6 and 9-14 are always the same but the
black, white and grays differ from "normal" video...
color0: black becomes light gray
color7: light gray becomes black
color8: dark gray becomes white
color15: white becomes dark gray.
Blech.

So at any rate, I brought this up because running task in the RHEL/CentOS/SL
xterm which is black on white looks like shit.  See attached.  With the
information above, I no longer think task incorrectly uses xterm colorization,
but rather the default (dark-256?) theme makes incorrect assumptions about
background and foreground colors.  That could (maybe should, I guess) be 
considered "user error".  

But on the other hand, new potential Taskwarrior users with RHEL/CentOS/SL 
(or even MacOX xterm, which is black on white?) may walk away thinking 
"this is crap--the colorization is broken".  I was  close to doing exactly 
that, IIRC, but I was motivated to figure out what was going on because 
I was very sick and tired of keeping TODO lists in plain text files.

**Comment by taskwarrior** [comment]
_steve rader on 2012-12-07T15:42:03Z says:_ 

 ...screenshot

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2012-12-07T15:46:39Z says:_ 

 So there are "dark" and "light" themes, the default is dark, and maybe it shouldn't be.  This seems to be the essence of this issue.

**Comment by taskwarrior** [comment]
_steve rader on 2012-12-07T16:04:20Z says:_ 

 
> So there are "dark" and "light" themes, the default is dark, and maybe it shouldn't be.  This seems to be the essence of this issue.

Yep.  Maybe.  I'm pretty sure RHEL, CentOS, SL, Ubuntu and MacOS-X all bring up
black on white xterms by default.  But CERTAINLY DON'T trust my memory!!
And also keep in mind, I have NO information or experiences with Taskwarrior in 
other terminal types.

**Comment by taskwarrior** [comment]
_David Patrick on 2012-12-07T20:11:09Z says:_ 

 Isn't there a programatic way to determine whether the default is "light" or "dark" ? 
If there is, couldn't taskwarrior serve up corresponding defaults?

**Comment by taskwarrior** [comment]
_steve rader on 2012-12-07T20:28:20Z says:_ 

 David Patrick wrote:
> Isn't there a programatic way to determine whether the default is "light" or "dark" ?

...certainly not a run-time. 
...maybe by way some wonky stuff at compile time,but that IMHO is a bad idea.

Here's an idea:
<pre>
$ task
A configuration file could not be found in /home/rader

Would you like a sample /home/rader/.taskrc created, so taskwarrior can proceed? (yes/no) yes

Does your terminal have a dark background or a light background? (dark/light) light
</pre>

Then depending on $TERM and the answer, set/include the appropriate theme in ~/.taskrc.

**Comment by taskwarrior** [comment]
_David Patrick on 2012-12-07T21:05:03Z says:_ 

 like it!

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2012-12-22T14:23:00Z says:_ 

 Not sure if this should be solved via README or code.  Not all distros put up light terminals by default, so we need something.

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2014-12-26T15:41:45Z says:_ 

 Updated task-color.5 to point out that when using a dark background terminal window, a dark theme gives better results.

There is no automated way to determine this.

**Comment by pbeckingham** [comment]
![tw-24_xterm-black-on-white-w-task-output](https://user-images.githubusercontent.com/7963/36635535-bd4d9e6a-1984-11e8-83f3-3c055fefbff9.gif)
![tw-24_xwd-27930](https://user-images.githubusercontent.com/7963/36635536-bd5b04a6-1984-11e8-87fa-87b7531982de.jpg)
![tw-24_xwd-27985](https://user-images.githubusercontent.com/7963/36635537-bd77b024-1984-11e8-8be8-1223592aebc9.jpg)
